solver/llbsolver/solver.go

```diff
@@ -135,6 +140,69 @@ func (s *Solver) Solve(ctx context.Context, id string, sessionID string, req fro
 
 	defer j.Discard()
 
+	var stopTrace func() []tracetest.SpanStub
+
+	if s := trace.SpanFromContext(ctx); s.SpanContext().IsValid() {
+		if exp, err := detect.Exporter(); err == nil {
+			if rec, ok := exp.(*detect.TraceRecorder); ok {
+				stopTrace = rec.Record(s.SpanContext().TraceID())
```

Comment:
We are gathering spans for the build and save them to the history now. `TraceID()` is used to be sure we capture the right traces for the build right?

Reply:
We filter out the tracing info only for the same traceID that was set in the tracing context of the solve function. So two solves get their own individual traces.